### PR TITLE
🐛 Use parametrize instead of creating replicated test functions

### PR DESCRIPTION
**Pull request by hibestil**
I've noticed there are recurring tests that can be collected within one parametrized test function. Using parameterized for them would be beneficial for creating a clean test environment.

### CODE REVIEW

File: platform/reworkd_platform/tests/agent/agent/remove_task_prefix_test.py [hunk 1]
```diff
@@ -1,79 +1,26 @@
...
+        ("Step: This is a sample task", "This is a sample task"),
+        ("Step 1: Perform a comprehensive analysis of system performance.", "Perform a comprehensive analysis of system performance."),
+        ("Step 2:Create a python script", "Create a python script"),
+        ("Step:This is a sample task", "This is a sample task"),
+        (". Conduct research on the history of Nike", "Conduct research on the history of Nike"),
+        (".This is a sample task", "This is a sample task"),
+        ("1. Research the history and background of Nike company.", "Research the history and background of Nike company."),
+    ],
+)
+def test_removes_task_colon(task_input, expected_output) -> None:
```
Comment (asim-shrestha) [inline platform/reworkd_platform/tests/agent/agent/remove_task_prefix_test.py:23]: lets change this to 

```suggestion
def test_removes(task_input, expected_output) -> None:
```

**Review by asim-shrestha — APPROVED** [review]

**Comment by asim-shrestha** [comment]
Awesome! Yeah our tests are quite lacking. Would love to make them better if you're interested

Looks good to merge, just has the one comment

File: platform/reworkd_platform/tests/agent/agent/remove_task_prefix_test.py (hunk 1 above)
Comment (hibestil) [reply platform/reworkd_platform/tests/agent/agent/remove_task_prefix_test.py:23]: I'd prefer to use a related function name with the test name. IMO, we should use `test_ remove_task_prefix` for the test name.

**Comment by hibestil** [comment]
@asim-shrestha I've made some changes according to your comments. Fyi.

**Comment by asim-shrestha** [comment]
Thanks @hibestil, we're going to be adding more tickets

Reach out if there's anything you want to tackle / need help with

**Pull request merged**